### DISCUSSION TITLE
How to add custom functions for Expression Language?

### DISCUSSION THREAD

**Discussion by fung** [gorules/zen#271] [Q&A] [answered]
Hi,

May I know is there a way to add custom function to [Expression Language](https://docs.gorules.io/docs/expression-language)? i.e. we would like to add some functions to support our operations by user, could be something like present/future value.

Thanks!

**Comment by ivanmiletic** [comment, marked as answer]
Hi Fung, at the moment it is not possible and we do not have immediate requests from our enterprise customers. If you have some functions in mind - please write them down and if we determine they are valuable we can implement them on our side

**Reply by fung** [reply]
Hi Ivan,

We are developing a lending system using zen-engine. While it is powerful and comprehensive already, one of most asked features is to allow risk team to use financial functions for money calculations.

I think the closest one we want is similar to [finance-math](https://crates.io/crates/finance-math).

Thanks for your consideration.
